Light Freshness Edit: ASP.NET - Security topics (#37129)
dotnet/AspNetCore.Docs: aspnetcore/security/data-protection/introduction.md

* A file path
* A permission
* A handle or other indirect reference
* Some server-specific data

Isolation is a requirement. Because modern apps are componentized, individual components want to take advantage of this system without regard to other components in the system. For instance, consider a bearer token component using this stack. It should operate without any interference, for example, from an anti-CSRF mechanism also using the same stack.

Some common assumptions can narrow the scope of requirements:

* All services operating within the cryptosystem are equally trusted.
* The data doesn't need to be generated or consumed outside of the services under our direct control.
* Operations must be fast because each request to the web service might go through the cryptosystem one or more times. The speed requirement makes symmetric cryptography ideal. Asymmetric cryptography isn't used until it's required.

## Design philosophy

ASP.NET Core data protection is an [easy to use](xref:security/data-protection/using-data-protection) data protection stack based on the following principles:

> * **Support ease of configuration**. The system strives for zero configuration. In situations where developers need to configure a specific aspect, such as the key repository, those specific configurations aren't difficult.

Reviewer: ```suggestion
* **Configuration should be easy**. The system strives for zero configuration. In situations where developers need to configure a specific aspect, such as the key repository, those specific configurations aren't difficult.
```

* **Offer basic consumer-facing APIs**. The APIs are straight forward to use correctly and difficult to use incorrectly.

> * **Don't require learning key management principles**. The system handles algorithm selection and key lifetime on behalf of the developer. Developers don't have access to the raw key material, so they don't need expert knowledge of the principles.

Reviewer: ```suggestion
* **Don't require the developer to learn the principles of managing keys**. The system handles algorithm selection and key lifetime on behalf of the developer. Developers don't have access to the raw key material, so they don't need expert knowledge of the principles.
```
